docs/articles/multi-attribute-semantic-search.md

```diff
@@ -0,0 +1,459 @@
+   - The results are fragmented, with different monsters being relevant for different attributes.
+
+This fragmentation in the results highlights a potential issue with our current search approach, as it fails to find monsters that satisfy all criteria at once.
+
+Maybe we can fix this issue by proactively retrieving more monsters for each attribute?
+Let's try it with 6 monsters for each attribute instead of 3.
+
+| id                  |   score_look |   score_habitat |   score_behavior |
+|:--------------------|-------------:|----------------:|-----------------:|
+| Whispering Shade    |     0.503578 |        0.609567 |                  |
+| Sandstorm Djinn     |     0.407344 |        0.365061 |                  |
+| Luminoth            |     0.378619 |                 |         0.345566 |
+| Nebula Jellyfish    |     0.36627  |                 |         0.259969 |
+| Dreamweaver Octopus |     0.315679 |                 |                  |
+| Quantum Firefly     |     0.288578 |                 |                  |
+| Fungal Network      |              |        0.438856 |                  |
+| Thornvine Elemental |              |        0.423421 |                  |
+| Mist Phantom        |              |        0.366816 |         0.236649 |
+| Stoneheart Golem    |              |        0.342287 |                  |
+| Living Graffiti     |              |                 |         0.385741 |
+| Crystalwing Drake   |              |                 |         0.385211 |
+| Aqua Wraith         |              |                 |         0.283581 |
+
+In total, we retrieved 13 monsters (that is more than half of our tiny dataset!) and still have the same issue: no monster was retrieved for all three attributes simultaneously.
+We could try to increase the number of monsters further, but it would result in the following shortcomings:
+1. While not a problem for our toy task, in production it will result in noticeably longer search time.
+2. With more attributes, the number of monsters to retrieve for every attribute will grow exponentially.
+3. And there is still no guarantee that this will result in retrieval of monsters that satisfy all criteria at once.
+
+This example is enough to show that the naive approach is not suitable for our task.
+
+### Superlinked approach
+
+Let's implement the second approach using Superlinked framework.
+
+First, let's define the schema, spaces, index and query:
+
+```python
+@schema
+class Monster:
+    id: IdField
+    look: String
+    habitat: String
+    behavior: String
+
+
+monster = Monster()
+
+look_space = TextSimilaritySpace(text=monster.look, model=MODEL_NAME)
+habitat_space = TextSimilaritySpace(text=monster.habitat, model=MODEL_NAME)
+behavior_space = TextSimilaritySpace(text=monster.behavior, model=MODEL_NAME)
+
+monster_index = Index([look_space, habitat_space, behavior_space])
+
+monster_query = (
+    Query(monster_index)
```

Comment:
space weights now default to 0 in the latest Superlinked. I suppose this query (and subsequent ones) need adjustment (to explicitly give `1` weights)

Reply:
@morkapronczay, could you please elaborate on this?
In the article I always specify all the weights explicitly.

Reply:
in this very example space weights are omitted.
```suggestion
    Query(monster_index, weights={look_space: Param("look_weight"), habitat_space: Param("habitat_weight"), behavior_space: Param("behavior_weight")})
```